### PR TITLE
Updatelesson

### PR DESCRIPTION
**Pull request by linda-seiter**
I created a new lesson for terminal operations (minus reduce and collect, which have separate existing lessons)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,30 +2,344 @@
...
+
+```java
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class Example {
+  public static void main(String[] args) {
+    List<Integer> ages = List.of(12, 55, 37, 9);
+    //List is not empty, a maximum value is returned as Optional[55]
```
Comment (kcleland0818) [inline README.md:70]: Just to add a little more readability, could we put a new line here in between `List<Integer> ages = ...` and the comment?

File: README.md [hunk 2]
```diff
@@ -2,30 +2,384 @@
...
+55
+Optional.empty
+max price unavailable
+```
+
+The `min` method is similar in that it takes a `Comparator` as
+parameter and returns an `Optional`, the minimum element in the stream.
+
+
+### Primitive stream max and min methods
```
Comment (kcleland0818) [inline README.md:161]: Not sure how often we even find ourselves using the min and max methods, but would we want to showcase using it with a data type that isn't as intuitive than a number? Like a LocalDate or even String?

File: README.md [hunk 3]
```diff
@@ -2,30 +2,384 @@
...
+import java.util.Arrays;
+import java.util.function.IntPredicate;
+import java.util.stream.IntStream;
+
+public class ExampleIntPredicateMatch {
+    public static void main(String[] args) {
+        int[] grades = {90, 75, 100, 58};
+        IntPredicate isPassing = i -> i >= 60;
+        IntStream gradesStream = Arrays.stream(grades);
+        System.out.println("All at least 18? " + gradesStream.allMatch(isPassing));  //false
```
Comment (kcleland0818) [inline README.md]: I think "All at least 18?" should really be "All passing?" (or something to that affect)

File: README.md [hunk 4]
```diff
@@ -2,30 +2,384 @@
...
+        //Create another stream
+        greetStream = Arrays.stream(greetings);
+        greetStream.forEach(System.out::println);  //hi  hello  howdy
+    }
+}
+```
+
+## reduce and collect
+
+The `reduce` and `collect` terminal operations are explained in separate lessons.
```
Comment (kcleland0818) [inline README.md]: Should we even mention this here then?

**Review by kcleland0818 — COMMENTED** [review]
Just a few small comments - great addition adding this lesson into the curriculum!

File: README.md (hunk 4 above)
Comment (linda-seiter) [reply README.md]: ok I removed this

File: README.md (hunk 1 above)
Comment (linda-seiter) [reply README.md:70]: ok, I added blank line.

File: README.md (hunk 2 above)
Comment (linda-seiter) [reply README.md:161]: Good suggestion.  I added an example with an Employee class and comparators for age and salary to find min and max.  Does that work?

File: README.md (hunk 3 above)
Comment (linda-seiter) [reply README.md]: thanks! I fixed the print statement

**Review by AlveeM — APPROVED** [review]

**Pull request merged**